Lib/posixpath.py

```diff
@@ -262,6 +262,9 @@ def expanduser(path):
             # password database, return the path unchanged
             return path
         userhome = pwent.pw_dir
+    # if the current user has no home directory, return the path unchanged
+    if userhome is None:
```

Comment:
I'm not comfortable to change the behavior on any platform.

I would be ok if the change is specific to VxWorks, if you add:

```suggestion
    if userhome is None and sys.platform == "vxworks":
```

Please update the comment and the NEWS entry to mention that only VxWorks is impacted.

Reply:
modified.